java/src/main/java/com/microsoft/graph/bulk/impl/model/GremlinEdge.java

```diff
@@ -0,0 +1,51 @@
+package com.microsoft.graph.bulk.impl.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Data
+@Builder
+@AllArgsConstructor
+@EqualsAndHashCode
+public class GremlinEdge {
+    private String id;
+    private String label;
+
+    private GremlinEdgeVertexInfo destinationVertexInfo;
+
+    private GremlinEdgeVertexInfo sourceVertexInfo;
+
+    private GremlinPartitionKey partitionKey;
+
+    //TODO: Evaluate impacts to changing this to be a HashMap<String, Serializable>
+    private HashMap<String, Object> properties;
+
+    public GremlinEdge() {
+        id = UUID.randomUUID().toString();
+        destinationVertexInfo = new GremlinEdgeVertexInfo();
+        sourceVertexInfo = new GremlinEdgeVertexInfo();
+        properties = new HashMap<>();
+    }
+
+    public void addProperty(String key, Object value) {
+        properties.put(key, value);
+    }
+
+    public void putProperties(Map<String, Object> properties) {
+        this.properties.putAll(properties);
+    }
+
+    //TODO: expand the validation logic to include id and label
```

Comment:
Is this TODO still valid?

Reply:
Added additional logic here as well as in the Vertex and PartitionKey